x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MlInitializationService.java

```diff
@@ -143,8 +144,13 @@ public void clusterChanged(ClusterChangedEvent event) {
                 event.state(),
                 MasterNodeRequest.DEFAULT_MASTER_NODE_TIMEOUT,
                 ActionListener.wrap(r -> isIndexCreationInProgress.set(false), e -> {
+                    if (e.getMessage().equals(previousException)) {
```

Comment:
Note: A bit fancier (but atomic and a bit shorter) way of writing this would be:

```
AtomicReference<String> previousException = new AtomicReference<>();
if (e.getMessage().equals(previousException.getAndSet(e.getMessage()))) {
  logger.debug(...);
} else {
  logger.error(...);
}
```